File: backend/src/main/java/com/inventory/backend/repository/InventoryAssetMovementRepository.java
package com.inventory.backend.repository;

import com.inventory.backend.entity.InventoryAssetMovement;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface InventoryAssetMovementRepository extends JpaRepository<InventoryAssetMovement, Long> {
    List<InventoryAssetMovement> findAllByOrderByMovedAtDesc();

    List<InventoryAssetMovement> findByAssetIdOrderByMovedAtDesc(Long assetId);
}

File: backend/src/main/kotlin/com/inventory/backend/repository/InventoryAssetMovementRepository.kt
package com.inventory.backend.repository

import com.inventory.backend.entity.InventoryAssetMovement
import org.springframework.data.jpa.repository.JpaRepository

interface InventoryAssetMovementRepository : JpaRepository<InventoryAssetMovement, Long> {
    fun findAllByOrderByMovedAtDesc(): List<InventoryAssetMovement>
    fun findByAssetIdOrderByMovedAtDesc(assetId: Long): List<InventoryAssetMovement>
}
